File: Fortran/ex2/costFunction.f90
	subroutine costFunction(m, theta, X, y, JCost, grad)
	implicit none
	integer::i, m
	double precision,dimension(2, m) :: X
	integer,dimension(m) :: y
	double precision,dimension(3) :: theta, grad
	double precision :: JCost
	double precision :: sigmoid, g
		
	
	do i = 1, m
		g = sigmoid( theta(1)+ theta(2)*X(1,i) + theta(3)*X(2,i))
		JCost = JCost - y(i) * log(g)- (1- y(i))* log(1- g)
		
		grad(1) = grad(1) - 1.0000*(g -y(i))
		grad(2) = grad(2) - X(1,i)*(g -y(i))
		grad(3) = grad(3) - X(2,i)*(g -y(i))
	enddo
	
	JCost = JCost/m
	grad = grad/m
	
	end

File: C/ex2/costFunction.c
#include <stdio.h>
#include <math.h>
double sigmoid(double z);

double costFunction(int M, int N, double theta[N], double X[M][2], double y[M], double grad[N])
{
	int i ;
	double   g, JCost = 0;
		
	
	for(i=0; i < M; i++){
		g = sigmoid( theta[0]+ theta[1]*X[i][0] + theta[2]*X[i][1]);
		JCost = JCost - y[i]*log(g) - (1- y[i])* log(1- g);
		
		grad[0] = grad[0] - 1.00000*(g - y[i]);
		grad[1] = grad[1] - X[i][0]*(g - y[i]);
		grad[2] = grad[2] - X[i][1]*(g - y[i]);
	}
	
	JCost = JCost/M;
	
	for(i=0; i < N; i++)
	       grad[i] = grad[i]/M;
	
	printf("%lf, %lf \n", grad[0], JCost);
//		getchar();
	return JCost;
}
